src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs

```diff
@@ -388,39 +383,37 @@ public static void ComputeUIntsToUint(Span<uint> input, Span<uint> output)
 
         public static uint[] ComputeUIntsToUint(Span<uint> input, int size)
         {
-            var output = new uint[size / 4];
+            uint[] output = new uint[size / sizeof(uint)];
             ComputeUIntsToUint(input, output);
             return output;
         }
 
         public static uint[] ComputeBytesToUint(byte[] input, int size)
         {
-            var output = new uint[size / 4];
+            uint[] output = new uint[size / sizeof(uint)];
             ComputeHash(input, MemoryMarshal.Cast<uint, byte>(output));
             return output;
         }
 
-        // compute a keccak hash (md) of given byte length from "in"
+        // compute a Keccak hash (md) of given byte length from "in"
         public static void ComputeHash(ReadOnlySpan<byte> input, Span<byte> output)
         {
-            var size = output.Length;
-            var roundSize = GetRoundSize(size);
-            var roundSizeU64 = GetRoundSizeU64(size);
+            int size = output.Length;
+            int roundSize = GetRoundSize(size);
             if (output.Length <= 0 || output.Length > STATE_SIZE)
             {
-                throw new ArgumentException("Bad keccak use");
+                ThrowBadKeccak();
             }
 
             Span<ulong> state = stackalloc ulong[STATE_SIZE / sizeof(ulong)];
             Span<byte> temp = stackalloc byte[TEMP_BUFF_SIZE];
 
-            var remainingInputLength = input.Length;
-            int i;
-            for (; remainingInputLength >= roundSize; remainingInputLength -= roundSize, input = input.Slice(roundSize))
+            int remainingInputLength = input.Length;
+            for (; remainingInputLength >= roundSize; remainingInputLength -= roundSize, input = input[roundSize..])
             {
-                var input64 = MemoryMarshal.Cast<byte, ulong>(input);
+                ReadOnlySpan<ulong> input64 = MemoryMarshal.Cast<byte, ulong>(input[..roundSize]);
 
-                for (i = 0; i < roundSizeU64; i++)
+                for (int i = 0; i < input64.Length; i++)
```

Comment:
Isn't it slightly faster if we store `input64.Length` in a local variable instead of accessing it on each iteration?

Reply:
Yes and no. For a `List<T>` or interface type then it would be.

However for a local reference of an array (assign array to local variable) or `Span<T>` the Jit recognises the `.Length` pattern and then eliminates all the out of bounds checks on the accesses in the for loop, so it ends up being faster.